src/__tests__/acceptance/audit.mixin.acceptance.ts

```diff
@@ -0,0 +1,755 @@
+};
+
+describe('Audit Mixin', () => {
+  const testDataSourceInstance = new TestDataSource();
+  const getCurrentUser = sinon.stub().resolves({
+    id: 'testId',
+    name: 'testName',
+  });
+  const auditLogRepositoryInstance = new TestAuditLogRepository(
+    new TestAuditDataSource(),
+  );
+  const getAuditLogRepository = sinon
+    .stub()
+    .resolves(auditLogRepositoryInstance);
+  const testRepositoryInstance = new TestRepository(
+    testDataSourceInstance,
+    getCurrentUser,
+    getAuditLogRepository,
+  );
+
+  it('should create audit log when new item is created', async () => {
+    //delete all entries in audit log
+    await deleteAll();
+
+    const mockItem = getMockItem();
+    const createMethodResponse = await testRepositoryInstance.create(mockItem);
+    const auditLog = await auditLogRepositoryInstance.find();
+    const storedRecords = await testRepositoryInstance.find({
+      where: {
+        id: mockItem.id,
+        description: mockItem.description,
+        itemName: mockItem.itemName,
+      },
+    });
+    expect(JSON.stringify(createMethodResponse)).to.equal(
+      JSON.stringify(mockItem),
+    );
+    expect(auditLog.length).to.equal(1);
+    expect(auditLog[0].actor).to.be.equal('testId');
+    expect(auditLog[0].entityId).to.be.equal(mockItem.id);
+    expect(JSON.stringify(auditLog[0].after)).to.be.equal(
+      JSON.stringify(mockItem),
+    );
+    expect(auditLog[0].before).to.be.undefined();
+    expect(auditLog[0].actionKey).to.be.equal(testAuditOpts.actionKey);
+    expect(auditLog[0].actedAt).to.be.Date();
+    expect(auditLog[0].actedOn).to.be.equal(TestModel.name);
+    expect(auditLog[0].action).to.be.equal(Action.INSERT_ONE);
+
+    //check if stores in db
+    expect(storedRecords.length).to.equal(1);
+  });
+
+  it('should not create audit log when options.noAudit is set to true on creating new item', async () => {
+    //delete all entries in audit log
+    await deleteAll();
+    const mockItem = getMockItem();
+    const createMethodResponse = await testRepositoryInstance.create(mockItem, {
+      noAudit: true,
+    });
+    expect(JSON.stringify(createMethodResponse)).to.equal(
+      JSON.stringify(mockItem),
+    );
+    const storedRecords = await testRepositoryInstance.find({
+      where: {
+        id: mockItem.id,
+        description: mockItem.description,
+        itemName: mockItem.itemName,
+      },
+    });
+
+    const auditLog = await auditLogRepositoryInstance.find();
+
+    expect(auditLog.length).to.equal(0);
+    //check if stores in db
+    expect(storedRecords.length).to.equal(1);
+  });
+
+  it('should create audit log when new items are created on calling createAll', async () => {
+    //delete all entries in audit log
+    await deleteAll();
+    const mockItemArray = getMockItemArray();
+    const createMethodResponse = await testRepositoryInstance.createAll(
+      mockItemArray,
+    );
+    expect(JSON.stringify(createMethodResponse)).to.equal(
```

Comment:
Because should.equal is the same as Node.js standard assert.equal, it will return false even on doing expect({}).to.equal({}). We need to do JSON.stringify to compare objects/Arrays.